### DISCUSSION TITLE
Draw Things API integration

### DISCUSSION THREAD

**Discussion by Jeremy-Developer-Page** [open-webui/open-webui#6863] [General]
# Feature Request

## Important Notes

- **Before submitting a report**: Please check the Issues or Discussions section to see if a similar issue or feature request has already been posted. It's likely we're already tracking it! If you’re unsure, start a discussion post first. This will help us efficiently focus on improving the project.

- **Collaborate respectfully**: We value a constructive attitude, so please be mindful of your communication. If negativity is part of your approach, our capacity to engage may be limited. We’re here to help if you’re open to learning and communicating positively. Remember, Open WebUI is a volunteer-driven project managed by a single maintainer and supported by contributors who also have full-time jobs. We appreciate your time and ask that you respect ours.

- **Contributing**: If you encounter an issue, we highly encourage you to submit a pull request or fork the project. We actively work to prevent contributor burnout to maintain the quality and continuity of Open WebUI.

- **Bug reproducibility**: If a bug cannot be reproduced with a `:main` or `:dev` Docker setup, or a pip install with Python 3.11, it may require additional help from the community. In such cases, we will move it to the "issues" Discussions section due to our limited resources. We encourage the community to assist with these issues. Remember, it’s not that the issue doesn’t exist; we need your help!

Note: Please remove the notes above when submitting your post. Thank you for your understanding and support!

---

**Is your feature request related to a problem? Please describe.**
I can't use Automatic1111 and Comfyui on macOS because it spends more than 1 min, through Core ML Draw Things implementations it manages to have generation times as low as 30 seconds on basic M1 chips.
**Describe the solution you'd like**
This program that exists for macOS, iPadOS and iOS implements the API Server function and makes all functionality available via API (similar to Automatic1111. The implementation is not too complex since all API functions are available by connecting to the default localhost. Being fast and optimized for macOS and iPadOS could you implement it in your UI please?
**Describe alternatives you've considered**
I have already tried to speed up automatic1111 but you get less than excellent results due to not using the Neural Engine present in Apple Silicon chips, same thing for Comfyui so that is why I came up with this solution, please implement it so even mac users can enjoy the speed of an image generated by their own chatbot.
**Additional context**

**Comment by Jeremy-Developer-Page** [comment]
So there are any news about this feature request?

**Reply by iamronr** [reply]
I'm checking in and wondering the same thing...

**Comment by 10a5er** [comment]
Actually the server also exists headless, is FOSS and not bound to apple-devices.

Also ppl made already translation-nodes to use the DT-server with comfy, maybe there is a short-term solution which uses those with the comfy integration.

https://github.com/drawthingsai/draw-things-community

**Comment by lpetflo** [comment]
I did a quick [DrawThings -> Automatic1111 conversion bridge](https://github.com/lpetflo/drawthings-2-open-webui), which lets you select the Automatic1111 integration in Open WebUI and use DrawThings as the actual backend.
Currently it's only working for image generation (not for editing) and generation size has to be set by the admin manually.
Feel free to suggest features or add pull requests :)

Edit: Newest version also features varying the image size, but adding something like "If a image size or ratio is given, this must be included in the prompt!" does definitely help.
